protocol/binary.go

```diff
@@ -61,3 +61,30 @@ func (binaryProtocol) DecodeEnveloped(r io.ReaderAt) (wire.Envelope, error) {
 	e, err := reader.ReadEnveloped()
 	return e, err
 }
+
+func (b binaryProtocol) DecodeRequest(r io.ReaderAt) (wire.Value, error) {
```

Comment:
This probably deserves a nice description an potentially a background into why this is useful/necessary :)

Reply:
So it is done.